store/tikv/lock_resolver.go

```diff
@@ -146,6 +149,79 @@ func (lr *LockResolver) getResolved(txnID uint64) (TxnStatus, bool) {
 	return s, ok
 }
 
+// BatchResolveLocks resolve locks in a batch
+func (lr *LockResolver) BatchResolveLocks(bo *Backoffer, locks []*Lock, loc RegionVerID) (bool, error) {
+	if len(locks) == 0 {
+		return true, nil
+	}
+
+	lockResolverCounter.WithLabelValues("resolve").Inc()
+
+	var expiredLocks []*Lock
+	for _, l := range locks {
+		if lr.store.GetOracle().IsExpired(l.TxnID, l.TTL) {
+			lockResolverCounter.WithLabelValues("expired").Inc()
+			expiredLocks = append(expiredLocks, l)
+		} else {
+			lockResolverCounter.WithLabelValues("not_expired").Inc()
+		}
+	}
+	if len(expiredLocks) == 0 {
+		log.Error("This should not occur!")
+		return true, nil
+	}
+
+	lr.cntTxnStatus = 0
+	lr.costTxnStatus = 0
+	txnInfos := make(map[uint64]uint64)
+	count := 0
+	for _, l := range expiredLocks {
+		count = count + 1
+		status, err := lr.getTxnStatus(bo, l.TxnID, l.Primary)
```

Comment:
There will be many locks share the same primary key, i think here we can check if the transaction status have stored in `txnInfos`.

Reply:
good idea!

Reply:
In fact, `getTxnStatus` will try to find the txn status from a LRU cache inside...